Resources/public/js/models/ez-contentmodel.js

```diff
@@ -232,45 +232,69 @@ YUI.add('ez-contentmodel', function (Y) {
          * @param {Function} callback
          */
         loadLocations: function (options, callback) {
-            var api = options.api,
-                currentLocation = options.location;
+            var locations = [],
+                that = this;
 
-            api.getContentService().loadLocations(this.get('id'), function (error, response) {
-                var tasks = new Y.Parallel(),
-                    loadError = false,
-                    locations = [];
+            this._loadLocationsView(options, function (error, response) {
+                var hits;
 
                 if ( error ) {
                     callback(error, response);
                     return;
                 }
-                Y.Array.each(response.document.LocationList.Location, function (loc) {
+
+                hits = response.document.View.Result.searchHits.searchHit;
+
+                Y.Array.each(hits, function (hit) {
                     var location,
-                        end;
+                        currentLocation = options.location;
 
-                    if (currentLocation && currentLocation.get('id') === loc._href) {
+                    if (currentLocation && currentLocation.get('id') === hit.value.Location._href) {
                         locations.push(currentLocation);
                     } else {
-                        location = new Y.eZ.Location({id: loc._href});
-                        end = tasks.add(function (err) {
-                            if ( err ) {
-                                loadError = true;
-                                return;
-                            }
-                            locations.push(location);
-                        });
-
-                        location.load(options, end);
+                        location = new Y.eZ.Location();
+                        location.setAttrs(location.parse({document: hit.value}));
+                        locations.push(location);
                     }
-                });
+                }, that);
 
-                tasks.done(function () {
-                    callback(loadError, locations);
+                locations.sort(function (a, b) {
+                    return (a.get('depth') - b.get('depth'));
                 });
+
+                callback(error, locations);
             });
         },
 
         /**
+         * Loads locations of the content using a view. The result is the REST API view containing
+         * locations from the path.
+         *
+         * @method _loadLocationsView
+         * @protected
+         * @param {Object} options
+         * @param {Object} options.api the JS REST client instance
+         * @param {Function} callback
+         */
+        _loadLocationsView: function (options, callback) {
```

Comment:
the name is a bit misleading actually I even wonder if this protected method is necessary, I guess you could put everything in `loadLocations`